### PR TITLE
EVG-17543 Wire up job logs and raw log links

### PR DESCRIPTION
**Pull request by khelif96**
EVG-17543

### Description 
Wires up the job logs button for task logs and also wires up html and raw links. 
Refactored context a bit to include an object for log metadata. 

Updated `.env-cmdrc.json` https://privatebin.corp.mongodb.com/?dc14d7f6a2ad4edd#2xf1TK81rcJ17UdZy145VPNAn1iedw2Qacv5mKHipvXK

### Screenshots
image

### CODE REVIEW

File: src/components/DetailsOverlay/ButtonRow/index.tsx [hunk 1]
```diff
@@ -43,8 +99,60 @@ const ButtonRow: React.FC = () => {
       >
         {hasCopied ? "Copied!" : tooltipText}
       </Tooltip>
-      <Button leftGlyph={<Icon glyph="Export" />}>RAW</Button>
-      <Button leftGlyph={<Icon glyph="Export" />}>HTML</Button>
+      {jobLogsURL && (
```
Comment (minnakt) [inline src/components/DetailsOverlay/ButtonRow/index.tsx:102]: I wonder if we should just disable this button rather than hide it

**Review by minnakt — CHANGES_REQUESTED** [review]

File: src/components/DetailsOverlay/ButtonRow/index.tsx [hunk 2]
```diff
@@ -43,8 +99,60 @@ const ButtonRow: React.FC = () => {
...
+          justify="middle"
+          trigger={
+            <Button
+              as="a"
+              data-cy="job-logs-button"
+              href={jobLogsURL}
+              leftGlyph={<Icon glyph="Export" />}
+              target="_blank"
+            >
+              JOB LOGS
```
Comment (sophstad) [inline src/components/DetailsOverlay/ButtonRow/index.tsx:114]: [nit] I don't think we need to go caps-lock mode here
```suggestion
              Job Logs
```

File: src/components/DetailsOverlay/ButtonRow/index.tsx [hunk 3]
```diff
@@ -43,8 +99,60 @@ const ButtonRow: React.FC = () => {
...
+        trigger={
+          <Button
+            as="a"
+            data-cy="raw-log-button"
+            disabled={!rawLogURL}
+            href={rawLogURL}
+            leftGlyph={<Icon glyph="Export" />}
+            target="_blank"
+          >
+            RAW
```
Comment (sophstad) [inline src/components/DetailsOverlay/ButtonRow/index.tsx:133]: Similarly
```suggestion
            Raw
```

File: src/components/DetailsOverlay/ButtonRow/index.tsx [hunk 4]
```diff
@@ -43,8 +99,60 @@ const ButtonRow: React.FC = () => {
...
+          </Button>
+        }
+      >
+        Open Raw log in a new tab
+      </Tooltip>
+      <Tooltip
```
[the remark this answers is not in the document]
Comment (khelif96) [reply src/components/DetailsOverlay/ButtonRow/index.tsx:139]: I don't know if its worth working around esp considering on subsequent renders it aligns correctly. I'll raise the issue with the LG folks

File: src/pages/LogView/LoadingPage/index.tsx [hunk 5]
```diff
@@ -28,30 +28,36 @@ const LoadingPage: React.FC<LoadingPageProps> = ({ logType }) => {
     [slugs.execution]: execution,
   } = useParams();
   const dispatchToast = useToastContext();
-  const { ingestLines } = useLogContext();
+  const { ingestLines, setLogMetadata } = useLogContext();
 
   let url = "";
   switch (logType) {
```
[the remark this answers is not in the document]
Comment (khelif96) [reply src/pages/LogView/LoadingPage/index.tsx:36]: I only stored taskId, execution and buildId intentionally they will be used for the subheader links in https://jira.mongodb.org/browse/EVG-17534

File: src/constants/logURLTemplates.ts [hunk 6]
```diff
@@ -42,10 +49,12 @@ enum originToType {
 const getEvergreenTaskLogURL = (
   taskID: string,
   execution: string,
-  origin: keyof typeof originToType
+  origin: keyof typeof originToType,
+  options: { text?: boolean; html?: boolean }
```
[the remark this answers is not in the document]
Comment (minnakt) [reply src/constants/logURLTemplates.ts:53]: bump on `html` being unused! or maybe I am reading something wrong?

File: src/components/DetailsOverlay/ButtonRow/index.tsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (minnakt) [reply src/components/DetailsOverlay/ButtonRow/index.tsx:86]: possibly giving the tooltip a fixed width might fix this issue 🤔

File: src/pages/LogView/LoadingPage/index.tsx [hunk 7]
```diff
@@ -30,30 +31,63 @@ const LoadingPage: React.FC<LoadingPageProps> = ({ logType }) => {
     [slugs.execution]: execution,
   } = useParams();
   const dispatchToast = useToastContext();
-  const { ingestLines } = useLogContext();
+  const { ingestLines, setLogMetadata } = useLogContext();
```
Comment (minnakt) [inline src/pages/LogView/LoadingPage/index.tsx:34]: It seems like there's a console error coming from this file. It's possible you might have to set the metadata in the `useEffect` below instead

Screen Shot 2022-10-21 at 4 20 52 PM

**Review by minnakt — APPROVED** [review]
Besides the console error, it looks good to me!

File: src/pages/LogView/LoadingPage/index.tsx (hunk 7 above)
Comment (khelif96) [reply src/pages/LogView/LoadingPage/index.tsx:34]: Good catch! Fixed

**Pull request merged**